eras/shelley/impl/src/Cardano/Ledger/Shelley/Genesis.hs

```diff
@@ -129,6 +143,37 @@ emptyGenesisStaking =
       sgsStake = mempty
     }
 
+-- | We do not handle the full precision of NominalDiffTime.
+newtype NominalDiffTimeRounded = NominalDiffTimeRounded NominalDiffTime
```

Comment:
On a second thought. Let's switch this `newtype` around the [`Micro`](https://hackage.haskell.org/package/base-4.14.1.0/docs/Data-Fixed.html#t:Micro) type.
```suggestion
-- Unlike `NominalDiffTime` that supports `Pico` precision, this type only supports `Micro` precision
newtype NominalDiffTimeMicro = NominalDiffTimeMicro Micro
```

Also implementing and exporting these three functions:
```
toNominalDiffTimeMicro :: NominalDiffTime -> Maybe NominalDiffTimeMicro

toNominalDiffTimeMicroWithRounding :: NominalDiffTime -> NominalDiffTimeMicro

fromNominalDiffTimeMicro :: NominalDiffTimeMicro -> NominalDiffTime
```

This way the type will be correct b y construction, instead of it rounding at the time of serialization

Reply:
I have updated the PR with these additions! :raised_hands:

Reply:
It is still a newtype wrapper around the `NominalDiffTime`. The point is that `NominalDiffTime` is a newtype wrapper around [`Pico`](https://downloads.haskell.org/ghc/9.2.5/docs/html/libraries/base-4.16.4.0/Data-Fixed.html#t:Pico) and we want a type that is a wrapper around [`Micro`](https://downloads.haskell.org/ghc/9.2.5/docs/html/libraries/base-4.16.4.0/Data-Fixed.html#t:Micro)

Reply:
I don't know how I missed that detail. :face_exhaling:. I have pushed the latest changes now :raised_hands: